Documentation/specs/dotnet-extensions.md

```diff
@@ -0,0 +1,110 @@
+## Abstract
+
+.NET extensions allows for a decentralized assembly store with decentralized versioning and servicing policies. To allow for this decentralization, the runtime host and the extensions have to agree on a set of contracts. This document outlines how extensions fit into the NuGet based acquisition and the CLI's project model all the way through the runtime host and servicing of components in extensions.
+
+## Installation
+
+Extensions contain two main components:
+
+1. A policy file that specifies various probing paths.
+2. A set of assemblies in a known layout.
+
+Extensions can be installed to a machine wide location and also unzipped when machine wide locations are not accessible (CI jobs.)
+
+1. **Global Extensions**: Installed by default on any location on the machine - This is done by the developer of the extension. The policy file for the extension is placed in `ProgramFiles(x86)\dotnetextensions\{Extension Name}\{Arch}.ext.json`
+2. **Local Extension**: Installed in `ProgramFiles(x86)\dotnetextensions\{Extension Name}\` or unzipped under `dotnet\..\dotnetextensions\{Extension Name}`
+
+
+### Global Extensions
+
+* Extensions are installed in a system wide folder. Example: `C:\Program Files (x86)\Fabrikam\`.
+* The policy file should be placed in the following location: `C:\Program Files (x86)\dotnetextensions\Fabrikam.NETCore.Ext\x64.ext.json` with the following contents:
+
+```
+{
+    "paths": {
+        "C:\Program Files (x86)\Fabrikam\x64": {
+            "layout": "nupkg",
+            "applyPatches": false,
+            "pathType": "absolute"
+        },
+        "C:\Program Files (x86)\Fabrikam\pkgs": {
+            "layout": "nupkg",
+            "applyPatches": false,
+            "pathType": "absolute"
+        }
+    }
+}
+```
+
+The runtime host probes the paths in the order they are specified in the `paths` section. On windows, the policy files have to be placed in `ProgramFiles(x86)\dotnetextensions` for 32-bit processes and for 64-bit processes. The supported architectures are: `x86` and `x64`.
+
```

Comment:
Should allow `ExpandEnvironmentStrings` or `wordexp`? So developers don't have to figure out where `ProgramFiles` actually is instead of `C:\`, so that they can specify global paths.